src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,206 @@
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+    
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.id as car_id, model, m.id as manufacturer_id, name, country "
+                + "FROM cars c JOIN manufacturers m ON c.manufacturer_id = m.id "
+                + "WHERE c.deleted = FALSE;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCar(resultSet));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a list of cars from carsDB.",
+                    throwable);
+        }
+        for (Car car : cars) {
+            car.setDrivers(getDriversForCar(car.getId()));
+        }
+        return cars;
+    }
+    
+    @Override
+    public Car update(Car car) {
+        String query = "UPDATE cars SET manufacturer_id = ?, model = ? "
+                + "WHERE id = ? AND deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateDriverStatement = connection.prepareStatement(query)) {
+            updateDriverStatement.setLong(1, car.getManufacturer().getId());
+            updateDriverStatement.setString(2, car.getModel());
+            updateDriverStatement.setLong(3, car.getId());
+            updateDriverStatement.executeUpdate();
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't update " + car + " in DB.", throwable);
+        }
+        deleteDrivers(car);
+        insertDrivers(car);
+        return car;
+    }
+    
+    @Override
+    public boolean delete(Long carId) {
+        String query = "UPDATE cars SET deleted = TRUE WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement = connection.prepareStatement(query)) {
+            deleteCarStatement.setLong(1, carId);
+            return deleteCarStatement.executeUpdate() > 0;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't delete car with id " + carId, throwable);
+        }
+    }
+    
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String getAllCarsByDriverRequest = "SELECT c.id as car_id, m.id as manufacturer_id, "
+                + "model, name, country "
+                + "FROM cars c JOIN manufacturers m ON manufacturer_id = m.id "
+                + "JOIN cars_drivers cd ON c.id = cd.car_id WHERE cd.driver_id = ? "
+                + "AND c.deleted = FALSE;";
```

Comment:
let's check if driver is not deleted as well

Reply:
done